crates/sui-framework/packages/sui-framework/sources/coin.move

```diff
@@ -556,6 +634,11 @@ public fun create_regulated_currency<T: drop>(
         coin_metadata_object: object::id(&metadata),
         deny_cap_object: object::id(&deny_cap),
     });
+    dynamic_field::add(
+        &mut metadata.id,
+        RegulatedMarker(),
+        RegulatedInfo::Regulated { allow_global_pause: false, deny_cap_id: object::id(&deny_cap) }
```

Comment:
`allow_global_pause` can end up converted to true when migrating to v2, while the metadata wouldn't reflect. Isn't this a small vulnerability for protocols who might trust the metadata field?